packages/react-ui/components/Tabs/Tabs.tsx

```diff
@@ -139,23 +137,23 @@ export class Tabs extends React.Component<TabsProps> {
     this.listeners.forEach((cb) => cb());
   };
 
-  private switchTab = (id: string) => {
+  private switchTab = (id: T) => {
     const { onValueChange, value } = this.props;
     if (id !== value && onValueChange) {
       onValueChange(id);
     }
   };
 
-  private getTab = (id: string): Tab | null => {
+  private getTab = (id: T): Tab<T> | null => {
```

Comment:
Здесь давай применим интерфейс контекста, раз метод реализует именно его. Меньше дублирования:
```suggestion
  private getTab: TabsContextType<T>['getTab'] = (id: T)=> {
```

В других методах надо сделать так же.